onnx/compose.py

```diff
@@ -0,0 +1,256 @@
+        io_map,  # type: List[Tuple[Text, Text]]
+        name=None,  # type: Optional[Text]
+        doc_string=None,  # type: Optional[Text]
+        producer_name='onnx.compose',  # type: Optional[Text]
+        ir_version=None,  # type: Optional[int]
+):  # type: (...) -> ModelProto
+    """Combines two ONNX models into a single one.
+
+    The combined model is defined by connecting the specified set of outputs/inputs. Those inputs/outputs
+    not specified in the io_map argument will remain as inputs/outputs of the combined model.
+
+    Arguments:
+        m1 (ModelProto): First model
+        m2 (ModelProto): Second model
+        io_map (list of pairs of string): The pairs of names [(out0, in0), (out1, in1), ...]
+                                          representing outputs of the first graph and inputs of the second
+                                          to be connected
+        name (string): Optional name for the combined graph
+                       By default, the name is g1.name and g2.name concatenated with an undescore delimiter
+        doc_string (string): Optional docstring for the combined graph
+                             If not provided, a default docstring with the concatenation of g1 and g2 docstrings is used
+        producer_name (string): Optional producer name for the combined model. Default: 'onnx.compose'
+        ir_version (int): Optional target IR version. By default the highest IR version from the two graphs will be used
+                          The target IR version should be compatible with the Operator Set versions used by
+                          the models to be combined.
+    """
+    if type(m1) is not ModelProto:
+        raise ValueError("m1 argument is not an ONNX model")
+    if type(m2) is not ModelProto:
+        raise ValueError("m2 argument is not an ONNX model")
+
+    opset_import_map = {}  # type: MutableMapping[Text, int]
+    opset_imports = \
+        [entry for entry in m1.opset_import] + \
+        [entry for entry in m2.opset_import]
+
+    for entry in opset_imports:
+        if entry.domain in opset_import_map:
+            found_version = opset_import_map[entry.domain]
+            if entry.version != found_version:
+                raise ValueError(
+                    "Can't merge two models with different operator set ids for a given domain. "
+                    f"Got: {m1.opset_import} and {m2.opset_import}")
+        else:
+            opset_import_map[entry.domain] = entry.version
+
+    min_ir_version = helper.find_min_ir_version_for(
+        [entry for entry in opset_imports])
+    if ir_version is None:
+        ir_version = max(min_ir_version, max(
+            m1.ir_version, m2.ir_version))
+    if ir_version < min_ir_version:
+        raise ValueError(
+            f"IR version {ir_version} is not sufficient to support "
+            f"the models operator set ids: {opset_imports}")
+
+    graph = merge_graphs(m1.graph, m2.graph, io_map, name, doc_string)
```

Comment:
should populate  
1. producer_name and producer_version - Something like "onnx model merge utility", version 1?
2. domain and model_version - These can be user inputs

Merge m1 and m2 fields
1. metadata_props and functions

I dont think "training_info" is being used, we can drop this.

Reply:
I think I covered that. Please check I didn't miss anything